ui/app/components/global-search/control.js

```diff
@@ -0,0 +1,170 @@
+import Component from '@ember/component';
+import { tagName } from '@ember-decorators/component';
+import { task } from 'ember-concurrency';
+import EmberObject, { action, computed, set } from '@ember/object';
+import { alias } from '@ember/object/computed';
+import { inject as service } from '@ember/service';
+import { run } from '@ember/runloop';
+import Searchable from 'nomad-ui/mixins/searchable';
+import classic from 'ember-classic-decorator';
+
+const SLASH_KEY = 191;
+
+@tagName('')
+export default class GlobalSearchControl extends Component {
+  @service dataCaches;
+  @service router;
+  @service store;
+
+  searchString = null;
+
+  constructor() {
+    super(...arguments);
+
+    this.jobSearch = JobSearch.create({
+      dataSource: this,
+    });
+
+    this.nodeSearch = NodeSearch.create({
+      dataSource: this,
+    });
+  }
+
+  keyDownHandler(e) {
+    const targetElementName = e.target.nodeName.toLowerCase();
+
+    if (targetElementName != 'input' && targetElementName != 'textarea') {
+      if (e.keyCode === SLASH_KEY) {
+        e.preventDefault();
+        this.open();
+      }
+    }
+  }
+
+  didInsertElement() {
+    this.set('_keyDownHandler', this.keyDownHandler.bind(this));
+    document.addEventListener('keydown', this._keyDownHandler);
+  }
+
+  willDestroyElement() {
+    document.removeEventListener('keydown', this._keyDownHandler);
+  }
+
+  @task(function*(string) {
+    try {
+      set(this, 'searchString', string);
+
+      const jobs = yield this.dataCaches.fetch('job');
+      const nodes = yield this.dataCaches.fetch('node');
+
+      set(this, 'jobs', jobs.toArray());
+      set(this, 'nodes', nodes.toArray());
+
+      const jobResults = this.jobSearch.listSearched;
+      const nodeResults = this.nodeSearch.listSearched;
+
+      return [
+        {
+          groupName: `Jobs (${jobResults.length})`,
+          options: jobResults,
+        },
+        {
+          groupName: `Clients (${nodeResults.length})`,
+          options: nodeResults,
+        },
+      ];
+    } catch (e) {
+      // eslint-disable-next-line
+      console.log('exception searching', e);
+    }
+  })
+  search;
+
+  @action
+  open() {
+    if (this.select) {
+      this.select.actions.open();
+    }
+  }
+
+  @action
+  selectOption(model) {
+    const itemModelName = model.constructor.modelName;
+
+    if (itemModelName === 'job') {
+      this.router.transitionTo('jobs.job', model.name);
```

Comment:
I think if this includes the namespace query param then the second bug I noticed goes away. I believe this is already being done in the allocation row template?